### DISCUSSION TITLE
Red life mechanic could be separated from Tag mode

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1168] [Other]
Case in point: games like Darkstalkers, Street Fighter 4 or 5, etc, where 1 on 1 matches have recoverable life.

If someone wants to have a move cause recoverable damage on the opponent in single modes, there is currently no direct way to do it as the recoverable life mechanic is locked to tag.

I guess one way to do this would be to make inflicted red life default to 0 outside of Tag mode, but still allow the engine to process red life and its recovery in other modes.

(This is all of course no issue in a full game)

**Comment by potsmugen** [comment]
I was thinking about this and maybe red life, dizzy and guard break could be configurable by team mode. Or enabling red life enables it in all modes.

**Reply by djonskin** [reply]
I think the first solution for Singles, etc. is better off on an author case-by-case basis. Triggers/scontrollers could better elaborate by example redlife uses in code outside of Tag. In broad strokes, I believe redlife is only common to Tag games, and a little more comeback mechanic-like elsewhere. Streets of Rage 4 experimented with it, so there has to be some favorable traction to gray/potential life sliver recovery mechanics.
